### PR TITLE
Blog post for ingress-nginx stabilization discussion

### PR DESCRIPTION
**Pull request by strongjz**
In an effort to keep the community up to date with the progress of Ingress-NGINX project, we are putting together a support plan for Kubernetes 1.22 and previous releases. This blog posts details that and how to solicit more support from the community.

### CODE REVIEW

**Comment by tao12345666333** [comment]
:+1: 

Should the impact on users be highlighted?  For example, the later version is no longer compatible with the old version of k8s.

**Comment by strongjz** [comment]
@tao12345666333 Added a table for notes about versions for k8s, ingress and support notes.

**Review by tao12345666333 — COMMENTED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+Kubernetes Version  | Ingress-NGINX version | Support Notes
+:-------------------|:----------------------|:------------
+`1.22`              | TBD                   | On Going
+`1.21`              | `v0.46.0`             | On Going but only CVE and crashes
+`1.20`              | `v0.46.0`             | On Going but only CVE and crashes
+`1.19`              | `v0.46.0`             | Support will drop 6 months after 1.22 release
+{{< /table >}}    
+
+Because of the networking v1 update going GA in Kubernetes 1.22, `v0.46.0` will 
+not work with Kubernetes 1.22.
```
Comment (tao12345666333) [inline content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:39]: Should we bold? Or highlight?

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+layout: blog
+title: 'Updates to Ingress Nginx with network v1'
+date: 2021-05-30
+slug: update-with-ingress-nginx
+---
+
+**Authors:** James Strong, Richardo Katz 
+
+With all Kubernetes APIs, there is a process to creating, maintaining, and
+ultimately deprecating them once they become GA. The network API is no
```
Comment (sftim) [inline content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:11]: There's a network API? Is that different from NetworkPolicy?

I know there's `networking.k8s.io/v1`; is this what we mean?

**Review by sftim — COMMENTED** [review]
I think the focus of this article will be to announce the support levels for https://github.com/kubernetes/ingress-nginx/

I think these are: _(will soon) work with v1 Ingress from `networking.k8s.io`; if you want to keep using an older Ingress API, limitations apply_

Does that sound right?

**Comment by sftim** [comment]
I'm afraid that this article doesn't seem ready for publication to the Kubernetes blog.

/lgtm cancel

**Comment by sftim** [comment]
/reopen
Oops. Misclick.

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+[KEP-1453:Graduate Ingress API to GA.](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1453-ingress-api#122)
+
+During our last community meeting the group has decided to continue to 
+support Kubernetes versions older than 1.22 with ingress version 0.46.0 
+for six months after the release of Kubernetes 1.22 with only CVE and 
+crashes only fixes, on a rolling basis. There will be a separate branch and 
+release of Ingress-NGINX to support this very similar to how Kubernetes 
+project proper does. As we advance the Ingress-NGINX project, the mainline 
+will support Kubernetes 1.22 and above, similar to the maintainers of the 
+Kubernetes project as a whole.
```
Comment (onlydole) [inline content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:27]:
```suggestion
During our last community meeting, the group has decided to continue supporting Kubernetes versions older than 1.22 with Ingress version 0.46.0. Support for Ingress-NGINX will continue for six months after Kubernetes 1.22 is released. Any additional bug fixes and CVEs for Ingress-NGINX will be addressed on a need-by-need basis.

Ingress-NGINX will have separate branches and releases of Ingress-NGINX to support this model, mirroring the Kubernetes project process. Future releases of the Ingress-NGINX project will track and support the latest versions of Kubernetes.
```

Comment (onlydole) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:27]: This was a bit difficult to read and I added in a line break and some other edits. Please let us know if any items here are not correct!

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md (hunk 1 above)
Comment (onlydole) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:39]: I'd vote for bolding in this context instead of highlighting!

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Because of the networking v1 update going GA in Kubernetes 1.22, `v0.46.0` will 
+not work with Kubernetes 1.22. 
+
+# What do I need to do
+
+The team is currently in the process of upgrading ingress-nginx to support 
+the v1 migration, you can track the progress 
+[here](https://github.com/kubernetes/ingress-nginx/pull/7156).  This also 
+means the feature requests are being frozen until the v1 update is completed.
```
Comment (onlydole) [inline content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:46]:
```suggestion
[here](https://github.com/kubernetes/ingress-nginx/pull/7156).  Feature requests are being frozen until the v1 update is completed. 
```

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+* Update to the latest version of Ingress-NGINX, currently [v0.46.0](https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v0.46.0) 
+* After Kubernetes 1.22 is released, ensure you are using the latest version of 
+  Ingress-NGINX that supports v1 networking api.
+
+The community’s feedback and support in this effort is welcome. The
+Ingress-Nginx Sub-project regularly hold community meetings where we discuss
+this and other issues facing the project, meeting notes and details are
+[here](https://github.com/kubernetes/community/tree/master/sig-network)
+
+
```
Comment (onlydole) [inline content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:59]:
```suggestion
```

**Review by onlydole — CHANGES_REQUESTED** [review]
Added in some changes that help with readability and clarify some points, please let me know if there are any questions!

**Comment by sftim** [comment]
@onlydole thanks for reviewing!

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+With all Kubernetes APIs, there is a process to creating, maintaining, and
+ultimately deprecating them once they become GA. The network API is no
+different. With the release of Kubernetes 1.22, the network API goes from
+v1beta1 to v1, with v1beta is deprecated. This move has been in discussion
+for quite some time now in 2017 in
+[issue #43214](https://github.com/kubernetes/kubernetes/issues/43214),
+[2019 with 1.16 api deprecation](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/)
+, and more recently in
+[KEP-1453:Graduate Ingress API to GA.](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1453-ingress-api#122)
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:18]: We could also hyperlink to https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md (hunk 2 above)
Comment (strongjz) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:11]: Yes, networking.k8s.io/v1 . I can update the wording on that to add clarity.

Comment (sftim) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:11]: networking.k8s.io is actually an _API group_ not an API. Ingress, within that API group, is now GA.

Also: everything in in `networking.k8s.io/v1` is GA because `v1` represents, for Kubernetes a stable API version - no `beta` or `alpha`.

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+date: 2021-05-30
+slug: update-with-ingress-nginx
+---
+
+**Authors:** James Strong, Richardo Katz 
+
+With all Kubernetes APIs, there is a process to creating, maintaining, and
+ultimately deprecating them once they become GA. The network API is no
+different. With the release of Kubernetes 1.22, the network API goes from
+v1beta1 to v1, with v1beta is deprecated. This move has been in discussion
```
Comment (sftim) [inline content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:13]: Did you perhaps mean:
```markdown
different. With the release of Kubernetes 1.22, SomeResourceName graduates to stable.
```
?

Ingress, IngressClass and NetworkPolicy are all already GA so I'm not sure what's graduating in v1.22.

Comment (bowei) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:13]: Is the entire v1beta deprecated or just v1beta1/Ingress

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md (hunk 2 above)
Comment (bowei) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:11]: My comment as well -- we should be specific. v1beta1 itself can't be deprecated because it's a just a (group, version). It's the Ingress API resource that is moving between versions.

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md (hunk 7 above)
Comment (sftim) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md:13]: See https://kubernetes.io/docs/reference/using-api/deprecation-guide/ for details.

> The networking.k8s.io/v1beta1 API version of IngressClass will no longer be served in v1.22.
> The extensions/v1beta1 and networking.k8s.io/v1beta1 API versions of Ingress will no longer be served in v1.22. 

I think those are the relevant deprecations for the ingress-nginx code.

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
+---
+layout: blog
+title: 'Updates to Ingress Nginx with network v1'
+date: 2021-06-21
+slug: update-with-ingress-nginx
+---
+
+**Authors:** James Strong, Richardo Katz
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:8]:
```suggestion
**Authors:** James Strong, Ricardo Katz
```
?

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+layout: blog
+title: 'Updates to Ingress Nginx with network v1'
+date: 2021-06-21
+slug: update-with-ingress-nginx
+---
+
+**Authors:** James Strong, Richardo Katz 
+
+With all Kubernetes APIs, there is a process to creating, maintaining, and
+ultimately deprecating them once they become GA. The networking.k8s.io API is no
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:11]:
```suggestion
ultimately deprecating them once they become GA. The networking.k8s.io API group is no
```

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md (hunk 7 above)
Comment (strongjz) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md]: updated using this wording.

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+different. With the release of Kubernetes 1.22, The networking.k8s.
+io/v1beta1 API version of IngressClass will no longer be served in 
+Kubernetes v1.22. The extensions/v1beta1 and networking.k8s.io/v1beta1 API 
+versions of Ingress also will no longer be served in v1.22.  This move has been
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:15]: How about:
```suggestion
different. The upcoming Kubernetes 1.22 release will remove several deprecated APIs
that are relevant to networking:

- the `networking.k8s.io/v1beta1` API version of [IngressClass](/docs/concepts/services-networking/ingress/#ingress-class)
- all beta versions of [Ingress](/docs/concepts/services-networking/ingress/): `extensions/v1beta1` and `networking.k8s.io/v1beta1`

On a v1.22 Kubernetes cluster, you'll be able to access Ingress and IngressClass
objects through the stable (v1) APIs, but access via their beta APIs won't be possible.
This change has been in
 ```
?

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+Kubernetes v1.22. The extensions/v1beta1 and networking.k8s.io/v1beta1 API 
+versions of Ingress also will no longer be served in v1.22.  This move has been 
+in discussion since
+[2017](https://github.com/kubernetes/kubernetes/issues/43214),
+[2019](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) with 
+1.16 Kubernetes API deprecations,and most recently in 
+KEP-1453: 
+[Graduate Ingress API to GA](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1453-ingress-api#122).
+
+During our community meetings, the group has decided to continue
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:23]: Is this correct:
```suggestion
During community meetings, the networking Special Interest Group has decided to continue 
```
?

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+versions of Ingress also will no longer be served in v1.22.  This move has been 
+in discussion since
+[2017](https://github.com/kubernetes/kubernetes/issues/43214),
+[2019](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) with 
+1.16 Kubernetes API deprecations,and most recently in 
+KEP-1453: 
+[Graduate Ingress API to GA](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1453-ingress-api#122).
+
+During our community meetings, the group has decided to continue 
+supporting Kubernetes versions older than 1.22 with Ingress version 0.47.0.
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:24]: There is no v0.47.0 version of the Ingress API. Did you mean Ingress-NGINX instead?

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+`1.22`              | TBD                   | On Going
+`1.21`              | `v0.47.0`             | On Going but only CVE and crashes
+`1.20`              | `v0.47.0`             | On Going but only CVE and crashes
+`1.19`              | `v0.47.0`             | Support will drop 6 months after 1.22 release
+{{< /table >}}    
+
+Because of the updates in Kubernetes 1.22, **v0.47.0** will not work with 
+Kubernetes 1.22. 
+
+# What do I need to do
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:46]:
```suggestion
# What you need to do
```

(“I” doesn't make sense for an article with two authors, and may especially confuse non-native speakers of English).

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+The team is currently in the process of upgrading ingress-nginx to support 
+the v1 migration, you can track the progress 
+[here](https://github.com/kubernetes/ingress-nginx/pull/7156).  
+Feature requests are being frozen until the v1 update is completed.
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:51]:
```suggestion
We're not making feature improvements to `ingress-nginx` until after the support for
Ingress v1 is complete.
```

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+# What do I need to do
+
+The team is currently in the process of upgrading ingress-nginx to support 
+the v1 migration, you can track the progress 
+[here](https://github.com/kubernetes/ingress-nginx/pull/7156).  
+Feature requests are being frozen until the v1 update is completed. 
+
+In the meantime to ensure no compatibility issues: 
+
+* Update to the latest version of Ingress-NGINX, currently
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:55]: punctuation nit:
```suggestion
* Update to the latest version of Ingress-NGINX; currently
```
(or put the current version details inside parens)

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+the v1 migration, you can track the progress 
+[here](https://github.com/kubernetes/ingress-nginx/pull/7156).  
+Feature requests are being frozen until the v1 update is completed. 
+
+In the meantime to ensure no compatibility issues: 
+
+* Update to the latest version of Ingress-NGINX, currently 
+  [v0.47.0](https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v0.47.0) 
+* After Kubernetes 1.22 is released, ensure you are using the latest version of 
+  Ingress-NGINX that supports v1 networking api.
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:58]:
```suggestion
  Ingress-NGINX that supports the stable APIs for Ingress and IngressClass.
```

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+The community’s feedback and support in this effort is welcome. The
+Ingress-Nginx Sub-project regularly hold community meetings where we discuss
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:61]:
```suggestion
Ingress-NGINX Sub-project regularly holds community meetings where we discuss
```
?

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md (hunk 1 above)
Comment (strongjz) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md]: bolded.

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+this and other issues facing the project, meeting notes and details are
+[here](https://github.com/kubernetes/community/tree/master/sig-network)
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:63]:
```suggestion
this and other issues facing the project. For more information on the sub-project, 
please see [SIG Network](https://github.com/kubernetes/community/tree/master/sig-network).
```

**Review by sftim — COMMENTED** [review]
Some more feedback

**Comment by sftim** [comment]
Maybe also hyperlink to https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/ ?

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md (hunk 12 above)
Comment (strongjz) [reply content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md]: yes, updated to clarify that point

File: content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md (hunk 2 above)
Comment (strongjz) [reply content/en/blog/_posts/2021-05-30-update-with-ingress-nginx.md]: updated.

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 19]
```diff
@@ -0,0 +1,69 @@
+---
+layout: blog
+title: 'Updates to Ingress Nginx with network v1'
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:3]: Maybe:
```suggestion
title: 'Updating NGINX-Ingress to use the stable Ingress API'
```
?

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+- the `networking.k8s.io/v1beta1` API version of [IngressClass](/docs/concepts/services-networking/ingress/#ingress-class)
+- all beta versions of [Ingress](/docs/concepts/services-networking/ingress/): `extensions/v1beta1` and `networking.k8s.io/v1beta1`
+
+On a v1.22 Kubernetes cluster, you'll be able to access Ingress and IngressClass
+objects through the stable (v1) APIs, but access via their beta APIs won't be possible.
+This change has been in
+in discussion since
+[2017](https://github.com/kubernetes/kubernetes/issues/43214),
+[2019](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) with 
+1.16 Kubernetes API deprecations,and most recently in
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:24]: nit:
```suggestion
1.16 Kubernetes API deprecations, and most recently in
```

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+releases of the Ingress-NGINX project will track and support the latest 
+versions of Kubernetes.
+
+{{< table caption="Ingress NGINX supported version with Kubernetes Versions" >}}
+Kubernetes Version  | Ingress-NGINX version | Support Notes
+:-------------------|:----------------------|:------------
+`1.22`              | TBD                   | On Going
+`1.21`              | `v0.47.0`             | On Going but only CVE and crashes
+`1.20`              | `v0.47.0`             | On Going but only CVE and crashes
+`1.19`              | `v0.47.0`             | Support will drop 6 months after 1.22 release
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:45]: Is this accurate?
```suggestion
v1.22              | ≥ v.0.47.0     | New features, plus bug fixes.
v1.21              | v0.47.x        | Bugfixes only, and just for security issues or crashes. No end-of-support date announced.
v1.20              | v0.47.x        | Bugfixes only, and just  for security issues or crashes. No end-of-support date announced.
v1.19              | v0.47.x        | Bugfixes only, and just  for security issues or crashes. Fixes only provided until 6 months after Kubernetes v1.22.0 is released.
```

I changed `v0.47.0` to `v0.47.x` because I assume there would be a new patch release, rather than yanking `v0.47.0` and replacing it with different code.

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+is released. Any additional bug fixes and CVEs for Ingress-NGINX will be 
+addressed on a need-by-need basis.
+
+Ingress-NGINX will have separate branches and releases of Ingress-NGINX to 
+support this model, mirroring the Kubernetes project process. Future 
+releases of the Ingress-NGINX project will track and support the latest 
+versions of Kubernetes.
+
+{{< table caption="Ingress NGINX supported version with Kubernetes Versions" >}}
+Kubernetes Version  | Ingress-NGINX version | Support Notes
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:40]:
```suggestion
Kubernetes version  | Ingress-NGINX version | Notes
```

**Review by sftim — COMMENTED** [review]
Overall, should this article say something about what folks should do if they are the author of a different Ingress controller that's not yet compatible with v1-only Ingress API?

**Comment by strongjz** [comment]
@sftim discussing with @rikatz  the focus for this blog post is to alert users about the deprecation of versions older than 1.19 in the new Nginx ingress controller version.

**Comment by sftim** [comment]
What I might do then is keep this post largely as is but also prepare a generic article about Ingress being deprecated, and put in a little bit of text that ties the two articles together.

The generic article might start life on https://k8s.dev/ and be targeted at people / organizations that are writing an Ingress controller and haven't yet announced policy for _their_ product and ecosystem around this deprecation, or perhaps haven't even started work to support the GA Ingress API.

TL;DR: carry on! We'll fit around you.

**Comment by strongjz** [comment]
@sftim  Alright, let me know if there is anything we/me need to do to get this released. Thanks for all your help.

**Review by tao12345666333 — APPROVED** [review]
/lgtm

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md [hunk 23]
```diff
@@ -0,0 +1,71 @@
+---
+layout: blog
+title: 'Updating NGINX-Ingress to use the stable Ingress API'
+date: 2021-06-21
+slug: update-with-ingress-nginx
+---
```
Comment (sftim) [inline content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md:6]: How about we get this merged _as a draft_?
Then, when we want to publish it, it'll be a small diff. We can also accept PRs to keep the draft current during the period before we can get that published.

```suggestion
---
draft: true
layout: blog
title: 'Updating NGINX-Ingress to use the stable Ingress API'
date: 2021-06-21
slug: update-with-ingress-nginx
---
```

**Comment by sftim** [comment]
I'd like this to follow close on the article in https://github.com/kubernetes/website/pull/28694

**Review by sftim — COMMENTED** [review]
@strongjz would you be happy to update the front matter (and path) for this for a pencilled-in publication date of the 15th of July?

File: content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md (hunk 23 above)
Comment (strongjz) [reply content/en/blog/_posts/2021-06-21-update-with-ingress-nginx.md]: updated.

**Comment by strongjz** [comment]
@sftim updated the date and draft status.

**Comment by reylejano** [comment]
As @sftim mentioned, this blog is good to follow the [Kubernetes API and Feature Removals In 1.22: Here’s What You Need To Know](https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/) blog that was released last week.
/lgtm

**Comment by sftim** [comment]
Set for publication on Monday 2021-07-26

Echoing LGTM from https://github.com/kubernetes/website/pull/28182#issuecomment-883747482
/lgtm
/approve
